fingerprint/fingerprint.go

```diff
@@ -52,19 +54,14 @@ func (f Fingerprint) String() string {
 	return f.Hex()
 }
 
-func (f Fingerprint) DeriveRSASigningKey(secretKey []byte) (*rsa.PrivateKey, error) {
-	h := sha256.New()
-
-	// hash.Hash is documented to never return an error
-	_, _ = h.Write(f[:])
-	_, _ = h.Write(secretKey)
-
-	var seed [32]byte
-	copy(seed[:], h.Sum(nil))
-
-	rand := rand2.NewChaCha8(seed)
+// seed generates a key-derivation-function seed from the fingerprint and given secret key.
+// Generated seeds _must_ be kept private.
+func (f Fingerprint) seed(secretKey []byte) [32]byte {
+	return sha256.Sum256(append(f[:], secretKey...))
```

Comment:
```suggestion
	return sha256.Sum256(append(slices.Clip(f), secretKey...))
```
Can we use slices.Clip here just to be safe (if Fingerprint were to change to a slice, this will keep working).

Reply:
Fixed in https://github.com/Venafi/token-exchange/pull/8/commits/d1fa29020b2527d54b05694ef30b95c42b59031e